apps/www/_blog/2025-09-18-postgrest-13-release.mdx

```diff
@@ -0,0 +1,141 @@
+---
+title: PostgREST 13
+description: New features and changes in PostgREST version 13.
+author: steve_chavez,laurenceisla,avallete
+image: 2025-09-18-postgrest-13-release/cover-postgrest-13.png
+thumb: 2025-09-18-postgrest-13-release/cover-postgrest-13.png
+categories:
+  - postgres
+tags:
+  - postgres
+  - postgrest
+  - release-notes
+date: '2025-09-18'
+---
+
+PostgREST 13 is out! It comes with API and Observabilty improvements. In this post, we'll see what's new.
+
+## Spread To-Many relationships
+
+This new feature allows you to represent one-to-many and many-to-many relationships as flat JSON arrays.
+
+For example, if you have database similar to IMDB and you’d like to represent it as a hierarchical JSON structure for your frontend, like so:
+
+```json
+[
+  {
+    "title": "The Shawshank Redemption",
+    "actors": ["Tim Robbins", "Morgan Freeman"],
+    "genres": ["Drama"]
+  },
+  {
+    "title": "The Godfather",
+    "actors": ["Marlon Brando", "Al Pacino"],
+    "genres": ["Drama", "Crime"]
+  },
+  {
+    "title": "The Dark Knight",
+    "actors": ["Christian Bale", "Heath Ledger"],
+    "genres": ["Drama", "Crime", "Action"]
+  }
+]
+```
+
+You can now do it this way:
+
+```js
+const { data, error } = await supabase.from('titles').select(`
+    title:primary_title,
+    ...people(actors:primary_name),
+    ...genres(genres:name)
+  `)
+```
+
+The above `...people` is “spreading” the many-to-many relationship between `titles` and `people`, forming a flat array only consisting of the `primary_name` column. This flat array is then renamed to `actors`. We do a similar process for `genres` , which also forms a many-to-many relationship with `people`.
+
+You can see the data model used for this example on this [gist](https://gist.github.com/steve-chavez/93f7ae04b4323e1952710af7129b32cf). There are more details about this feature on the [official docs](https://docs.postgrest.org/en/v13/references/api/resource_embedding.html#spread-to-many-relationships).
+
+## Automatic tsvector convertion
+
+Previously you could only use the full text search operator on `tsvector` columns, now you can do it on `text` columns too:
+
+```js
+const { data, error } = await supabase.from('titles').textSearch('primary_name', `'god' & 'father'`)
+```
+
+This works because `text` columns will be automatically converted with `to_tsvector`.
```

Comment:
this is very cool steve

i imagine it will be quite slow since there is not pre-build index? should we still recommend they create a generated column/index if they are using this frequently?

Reply:
Thanks for catching that :pray: . Added a note mentioning an index.